docs/index.md

```diff
@@ -1,91 +1,121 @@
-
-But how do we test concurrency in our code? Testing tools and frameworks have not kept up with the
-pace of language innovation and so this is where `Coyote` comes to the rescue. When you use Coyote
-to test your programs you will find concurrency bugs that are very hard to find any other way.
-
-Coyote rewrites your `async tasks` in a way that allows the [coyote test](tools/testing.md) tool to
-control all the concurrency and locking in your program and this allows it to find bugs using
-intelligent [systematic testing](core/systematic-testing.md).
-
-For example, if you take the `BoundedBuffer.dll` from the above sample you can do the following:
-
-```
-coyote rewrite BoundedBuffer.dll
-coyote test BoundedBuffer.dll -m TestBoundedBufferMinimalDeadlock --iterations 100
-```
+  }
+
+  $(document).ready(function () {
+      video  = $("#shortintro")[0];
+      caption = $("#caption")[0];
+      caption.style.display="none";
+      video.ontimeupdate = function() { show_captions(video, caption); };
+  });
+
+</script>
+
+Coyote helps write powerful, expressive tests for your code. You can declare sources of
+non-determinism (such as timers, failures, etc.) as part of your tests. The Coyote testing tool can
+_systematically_ explore a large number of interleavings of concurrent operations as well as
+non-deterministic choices so that it covers a large set of behaviors in a very short time. This is
+different from _stress testing_. Coyote takes control of the concurrency so that it can manipulate
+every possible scheduling. With appropriate _mocking_, Coyote can also do this in "developer" mode
+on a single laptop with little or no dependence on the bigger production environment.
+
+Coyote is not a verification system. It does not use theorem proving to make correctness guarantees,
+instead it uses intelligent search strategies to drive systematic testing, based on deep
+understanding of concurrency primitives that you have used in your code. This approach has proven to
+work well for large production teams, including many teams in Microsoft Azure because it has a small
+barrier to entry with almost immediate benefits for those who adopt it.
+
+Coyote does not require that a team starts from scratch and rebuild their system. Often it is too
+expensive to start over. Instead Coyote can be adopted gradually, adding more and more structure
+around your _Coyote-aware_ code. The more of this structure you add the more benefit you get from
+Coyote, but it is certainly not an all or nothing proposition.
+
+So Coyote brings together elements of design, development and testing into an integrated package
+that works really well in the real world. See our [case
+studies](case-studies/azure-batch-service.md) for some great customer testimonials.
+
+## Supported programming models
+
+Coyote supports two main programming models:
+
+- [Asynchronous tasks](programming-models/async/overview.md), which follows the popular [task-based
```

Comment:
Should we remove the drop-in replacement text since we now ask people to always use binary rewriting (and eventually we can drop support for the drop-in replacement once the remaining internal customers move to rewriting?) Perhaps we can rephrase this whole paragraph to something like:

- [Asynchronous tasks](programming-models/async/overview.md), which implements the popular [task-based asynchronous pattern](https://docs.microsoft.com/en-us/dotnet/standard/asynchronous-programming-patterns/task-based-asynchronous-pattern-tap). This programming model is centered around the `System.Threading.Tasks.Task` .NET type, which represents an asynchronous operation that the programmer can coordinate using the `async` and `await` keywords of [C#](https://docs.microsoft.com/en-gb/dotnet/csharp/). To use Coyote, you do not need to change your production code! During testing, you can run Coyote [binary rewriting](programming-models/async/rewriting.md) as a post-build task to automatically instrument your tests and application, taking control of `Task` objects and related concurrency types from the Task Parallel Library. This is where the magic happens. Coyote controls the execution of each Coyote `Task` so that it can explore various different interleavings to find bugs.

Reply:
well the bigger rewrite of all async content to downplay ControlledTask is still a "TODO", so I'd prefer to push that to a future pull request.

Reply:
Yep, true ... I was thinking to mostly do this only for the home page, but yes, we can do that later